src/components/smart-table-content/table-head/table-head.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cn from 'classnames';
+import injectSheet from 'react-jss';
+import { Th, Tr, Thead, Icon } from '../../../';
+import styles from './table-head.styles';
+import { colShape, sortingLocalizationShape } from '../shapes';
+
+const initialSortOrder = 'desc';
+const reverseSortOrder = 'asc';
+
+const readableSort = {
+  asc: 'ascending',
+  desc: 'descending',
+};
+
+const nextSort = sortOrder => (sortOrder === initialSortOrder ? reverseSortOrder : initialSortOrder);
+class TableHead extends Component {
+  getAriaSort = key => {
+    const { sortOrder } = this.props;
+    return this.isSortedOn(key) ? readableSort[sortOrder] : 'none';
+  };
+
+  getNextSortDir = key => {
+    const { sortOrder } = this.props;
+    return this.isSortedOn(key) ? nextSort(sortOrder) : initialSortOrder;
+  };
+
+  isSortedOn = key => {
+    const { sortField } = this.props;
+    return sortField === key;
+  };
+
+  renderIcon(key) {
```

Comment:
all "render" class methods can and should be their own Components in the React way of thinking. The arguments and the props it accesses should be that Component's props. Feel free to put it in the same file even, to avoid cluttering. Our lint rules allows multiple components in the same file (but only one stateful).